### ISSUE TITLE
Please Certify and add the CastGarden host

### ISSUE THREAD

**Issue by Agorise** [Podcast-Standards-Project/Certification#5]
Please review CastGarden and let us know if you would like us to demo anything for you like a livestream podcast, a podcast with transcription, etc. 
 
**The name of your product/service:** CastGarden 
 
**A link to your logo (vector preferred):** 
![CastGarden logo](https://github.com/Podcast-Standards-Project/Certification/assets/31280836/e2b252e0-529f-42a5-a5dc-def037907bea) 
 
**Product/service category:** 
Audio-only podcasts example:  
- https://cast.garden/feeds/podcast/videos.xml?videoChannelId=37  
 
Video podcasts example: 
- https://cast.garden/feeds/podcast/videos.xml?videoChannelId=35 
 
Livestreams too! -- Just ping me and I will do a livestream podcast for you. :) 
 
On every channel page, you will see a little green rss icon on https://Cast.Garden 
Just click that icon and select Podcast RSS 2.0. 
 
CastGarden supports: 
 
App Type 
- blockchain
- open source
- hosting
- node hosting
- social links
- video
- website 
 
Supported Elements 
- Alternate Enclosure
- Block
- Funding
- GUID
- Images
- Integrity
- License
- Live
- Locked
- Medium
- Person
- Podping
- Podroll
- Search
- Social Interact
- Source
- Tracker
- Transcript
- Value
- Video
 
Platforms 
- fediverse (if possible) see: https://cast.garden/about/follows (and: https://github.com/Podcastindex-org/web-ui/issues/287) 
- hive 
- mobile 
- web 
 
**A short description of your service (one sentence):** 
CastGarden enables any Audio, Video or Livestream Podcaster to Monetize and stream out 4K quality P2P over Web3, where you get unmatched Syndication out to the Fediverse Network, Blockchain Communities and 1000's of apps and website syndicates. 
 
**Your contact information (email, twitter):** 
Please ping us anytime by email (agorise@pm.me), matrix/element (http://agorise.chat), twitter (agorise_world), or on zoom.

**Comment by Agorise** [comment]
![CastGarden logo full](https://github.com/Podcast-Standards-Project/Certification/assets/31280836/7a84f911-7f24-46db-9dab-4bcbf3e211e9)

**Comment by Agorise** [comment]
Hi @mijustin I was just wondering if there was anything else that we can do to help finish up our PSP Certification and listing? Please let me know if ya could, huuuge thanx :)

**Comment by Agorise** [comment]
Or @snookfin @cameronmoll if there is anything I can do to help just let me know :)

**Comment by CharlesWiltgen** [comment]
Hey @Agorise, I ran https://cast.garden/feeds/podcast/videos.xml?videoChannelId=33 through the validator and have some notes. _(Caveat: I just completely rewrote the Podbase validator and added new tests for Podcast Standards Project, so it's possible that some of these are missed edge cases.)_

- The validator's PSP section reports **“Podcast” namespace was not found**, but that error message isn't quite right — there is a `podcast` namespace definition, but it's seemingly incorrect. It's currently `xmlns:podcast="https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md"`, but should be `xmlns:podcast="https://podcastindex.org/namespace/1.0"`.

- The validator's PSP section reports missing required `channel` elements.

- The validator notes a bunch of issues in the iTunes/Apple Podcasts section as well. Please let me know if you disagree with any of them. (Thanks!)

**Comment by Agorise** [comment]
Thanx for checkin it out @CharlesWiltgen :) 
Sorry you had to be the validator on my first attempt. I wish I had known ahead of time that that Podbase validator existed. Anyway, I will go over every red and yellow point that it made and see if I can tweak the feed generator in our code and then reply back here asap with the updated feed link (which gets auto-generated by every channel on the CastGarden site). Thanx again and chat soon :)

**Comment by CharlesWiltgen** [comment]
> Sorry you had to be the validator on my first attempt. I wish I had known ahead of time that that Podbase validator existed.

It's brand new! Honestly, I was very happy to be able to test a new feed source with it.

> …I will go over every red and yellow point that it made and see if I can tweak the feed generator in our code and then reply back here asap with the updated feed link

Great! And if the process turns up any validator problems I'll turn around fixes ASAP. Thank you!

**Comment by Agorise** [comment]
Ok @CharlesWiltgen I fixed just about everything you mentioned and then some: 
https://podba.se/validate/?url=https://cast.garden/feeds/podcast/videos.xml?videoChannelId=33 
 
Byte-range fyi is handled by the peertube server backend that we use so we cannot mess with that. CastGarden streams are P2P by the way so you get much better quality streams than any other host out there. 4K video and 320Kbps audio get broken up into chunks and streamed to your device from all the other seeders (listeners/watchers) out there of your podcast.  
 
File extension - One thing that I cannot change is the file extension since CastGarden uses m3u8 format via the peertube server software. M3u8 is more than just mp4 or mp3 files, it gives the player those formats yes, in multiple bitrate choices too, plus it includes playlist/tracks and other info. So m3u8 cannot be changed as it comes from the peertube server package itself that we have no control over. 
 
CastGarden already supports guid and transcript too. 
Guid we can't touch since that value is only generated occasionally by the peertube server backend. 
Transcript is not always filled unless the Podcaster uploads a transcript with their mp3/mp4. 
 
I hope that satisfies the requirements for our PSP Certification and you can maybe help us push this out there. 
If there is anything at all that we can do to help just let me know. 
Please and thank you again! :)

**Comment by CharlesWiltgen** [comment]
> _Byte-range fyi is handled by the peertube server backend that we use so we cannot mess with that._

**TLDR: Not a PSP-1 requirement.** This is an Apple requirement for streaming playback, but not a PSP-1 certification requirement. (If you can think of a way to make that or any other result clearer in the validator UI, feedback is always welcome.)

> *One thing that I cannot change is the file extension since CastGarden uses m3u8 format via the peertube server software.*

**TLDR: Not a PSP-1 requirement, but doesn't meet basic RSS requirements.** Apple Podcasts requires `<podcast:enclosure>` URLs end in a valid media extension, a requirement discovered by @jamescridland.

However, `m3u8` files are playlists, while `<podcast:enclosure>` elements link to audio or video files. To make valid podcast feeds, parse the PeerTube playlist for the primary episode file and use that for `<podcast:enclosure>`, then use `<podcast:alternateEnclosure>` for alternates.

> *CastGarden already supports guid…*

**TLDR: Not a PSP-1 requirement.** I see them for `<item>` elements, but not for `<channel>` elements as detected here.

> _CastGarden streams are P2P by the way so you get much better quality streams than any other host out there._

**TLDR: Unrelated to certification but…** FWIW, there's no intrinsic relationship between quality and whether media is delivered using P2P protocols or directly. However, PeerTube quality appears to top out at 4K, while services like YouTube, Vimeo, and others support 8K video.

My take is that you'll want to adapt PeerTube's use of playlists to instead use `<podcast:enclosure>` and `<podcast:alternateEnclosure>` elements, so your feeds will work reliably in podcast players. What do you think?

**Comment by Agorise** [comment]
Ok, I am going to see if I can parse the m3u8 playlist file that the server creates and extract the individual file links and further utilize the podcast:alternateEnclosure and podcast:enclosure elements. That shouldn't be too hard. Will try to knock that out this weekend, along with the import of the higher quality thumbnail images too. 
 
As for youtube and vimeo, yeah you're right, they offer 8K, but I wasn't thinking of them in this podcasting 2.0 web3 space at all. They are competitors in a way, but will never offer the open standards or syndication that we do :)  
 
Thanx again @CharlesWiltgen and I'll try to make these code mods asap and have them live on the CastGarden site by monday! :)

**Comment by Agorise** [comment]
Fyi @CharlesWiltgen we added some additional rss elements to the rss feeds so that CastGarden feeds and data are more accessible on more devices now. 
 
Channel id 33 rss feeds have been disabled, so here is a different example: 
https://validator.w3.org/feed/check.cgi?url=https://cast.garden/feeds/podcast/videos.xml?videoChannelId=37 
https://podba.se/?url=https://cast.garden/feeds/podcast/videos.xml?videoChannelId=37 
https://www.castfeedvalidator.com/validate.php?url=https://cast.garden/feeds/podcast/videos.xml?videoChannelId=37 
 
I hope you like it and can help us get our PSP Certification now :) 
Thanx so much and please let me know if there is anything we can do to help.

**Comment by Agorise** [comment]
Hey @CharlesWiltgen just curious, who should we contact about getting our PSP Certification for CastGarden? 
CastGarden is Podcasting 2.0 Certified as well, see: 
https://podcastindex.org/apps?appTypes=blockchain&elements=Live&platforms=Hive 
Thank you in advance for any help you can provide :)

**Comment by Agorise** [comment]
> ![CastGarden logo full](https://user-images.githubusercontent.com/31280836/251878454-7a84f911-7f24-46db-9dab-4bcbf3e211e9.svg) 
 
Thank you @CharlesWiltgen and @tomrossi7 for all your hard work! :) 
 
Q: Is there a Guide yet that shows us how to properly place the PSP logo on our site and what page on podstandards.org we should link it to? 
 
Ideally, it would be great if the CastGarden logo would show with the others on the homepage and we could just link right to that section, like this: https://podstandards.org/#Hosting-Providers

**Comment by Agorise** [comment]
Any news on this issue? 
https://github.com/Podcast-Standards-Project/Certification/issues/9 
We worked super hard to add all of the required rss elements and a lot more so that we could be shown as PSP Certified, so anything you guys can do to please help us get visible on the site now is greatly appreciated :)
